src/Numerics/DGMethods/DGModel.jl

```diff
@@ -1177,26 +1177,52 @@ function continuous_field_gradient!(
 
     if direction isa EveryDirection || direction isa VerticalDirection
         vertical_polyorder = N[dim]
-        vertical_D = grid.D[dim]
-        vertical_ω = grid.ω[dim]
-        event = kernel_continuous_field_gradient!(device, (Nq[1], Nq[2]))(
-            m,
-            Val(dim),
-            Val(N),
-            VerticalDirection(),
-            ∇state.data,
-            state.data,
-            grid.vgeo,
-            vertical_D,
-            vertical_ω,
-            Val(I),
-            Val(O),
-            # If we are computing in every direction, we need to
-            # increment after we compute the horizontal values
-            (direction isa EveryDirection);
-            ndrange = (nrealelem * Nq[1], Nq[2]),
-            dependencies = (event,),
-        )
+        if vertical_polyorder > 0
+            vertical_D = grid.D[dim]
+            vertical_ω = grid.ω[dim]
+            event = dgsem_auxiliary_field_gradient!(device, (Nq[1], Nq[2]))(
+                m,
+                Val(dim),
+                Val(N),
+                VerticalDirection(),
+                ∇state.data,
+                state.data,
+                grid.vgeo,
+                vertical_D,
+                vertical_ω,
+                Val(I),
+                Val(O),
+                # If we are computing in every direction, we need to
+                # increment after we compute the horizontal values
+                (direction isa EveryDirection);
+                ndrange = (nrealelem * Nq[1], Nq[2]),
+                dependencies = (event,),
+            )
+        else
+            Np = dofs_per_element(grid)
+            Nfp_v, Nfp_h = div.(Np, (Nq[1], Nq[end]))
+            nface = 2 * dim
```

Comment:
Instead of all this boiler plate at the top of the function where we compute all the relevant element/geometry info, can't we just do what we did when we split the DG kernels a while back and use the `basic_grid_info` and related functions? This can potentially avoid future issues if we can ensure kernels are getting values for `Np`, `Nfp_v`, etc. from the same source.

Reply:
Done